recipes/fmilib/all/conanfile.py

```diff
@@ -0,0 +1,197 @@
+
+    def requirements(self):
+        self.requires("fmi1/1.0.1")
+        self.requires("fmi2/2.0.4")
+        if self.version >= Version("3.0a2"):
+            self.requires("fmi3/3.0.1")
+        self.requires("expat/2.6.2")
+        self.requires("minizip/[>1.2.13 <=1.3.1]")
+        self.requires("zlib/[>=1.2.13 <2]")
+        # c99_snprintf -> should be externalised
+
+    def validate(self):
+        # https://github.com/modelon-community/fmi-library/issues/93
+        if self.settings.arch not in ["x86", "x86_64"] and Version(self.version).major < 3:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} does not support architecture "
+                f"'{self.settings.arch}' on {self.settings.os}")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+
+        tc = CMakeToolchain(self)
+        tc.variables["FMILIB_BUILD_STATIC_LIB"] = not self.options.shared
+        tc.variables["FMILIB_BUILD_SHARED_LIB"] = self.options.shared
+        tc.variables["FMILIB_BUILD_TESTS"] = self.options.with_fmus
+        tc.variables["FMILIB_FMI_STANDARD_HEADERS"] = posixpath.join(self.build_folder, "fmis").replace("\\", "/")
+        tc.variables["FMILIB_GENERATE_DOXYGEN_DOC"] = False
+
+        # The variable is an option only if the following condition is true
+        if not self.options.shared and not self.settings.os in ["Windows", "Macos"]:
+            tc.variables["FMILIB_BUILD_FOR_SHARED_LIBS"] = self.options.get_safe("fPIC", False)
+
+        if is_msvc(self):
+            tc.variables["FMILIB_BUILD_WITH_STATIC_RTLIB"] = is_msvc_static_runtime(self)
+
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0026"] = "OLD"
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0045"] = "OLD"
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0046"] = "OLD"
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
+
+        tc.generate()
+
+        cd = CMakeDeps(self)
+        cd.set_property("expat", "cmake_target_name", "expat")
+        cd.set_property("minizip", "cmake_target_name", "minizip")
+        cd.generate()
+
+        # This is needed if with_fmus=True
+        vre = VirtualRunEnv(self)
+        vre.generate(scope="build")
+
+        minizip_code = {
+            "1.3.1":
+            {
+                "url": "https://zlib.net/fossils/zlib-1.3.1.tar.gz",
+                "sha256": "9a93b2b7dfdac77ceba5a558a580e74667dd6fede4585b91eefb60f03b72df23"
+            },
+            "1.2.13":
+            {
+                "url": "https://zlib.net/fossils/zlib-1.2.13.tar.gz",
+                "sha256": "b3a24de97a8fdbc835b9833169501030b8977031bcb54b3b3ac13740f846ab30"
+            }
+        }
+        get(self,
+            **minizip_code[str(self.dependencies["minizip"].ref.version)],
+            pattern="*/minizip/*",
+            strip_root=True, destination=path.join(self.build_folder))
```

Comment:
Afaik, the library needs the actual source code for minizip instead of relying in the compiled library (Would appreciate insight into this too!)

You actually get access to the dependency's conandata thru the `self.dependencies` accessor, so this could be simplified to something like
 ```suggestion
        minizip_version = self.dependencies["minizip"].ref.version
        get(self,
            **self.dependencies["minizip"].conandata["sources"][minizip_version],
            pattern="*/minizip/*",
            strip_root=True, destination=path.join(self.build_folder))
```

which allivieates some of your concerns - I would still like some way to avoid this, but alas seems like this will be that way to go forward

Reply:
`fmilib` introduce `minizip.c` and `minunz.c` as functions instead of applications by changing their `main` functions. To do this, `fmilib` needs the `minizip` sources. It also uses other functionality in `minizip`, so the precompiled library is also needed. In patch file `*002*.patch`, you can see the new header files for the patched functions.